lisa/env.py

```diff
@@ -267,61 +267,63 @@ def from_cli(cls, argv=None):
         to be confusing (help message woes, argument clashes...), so for now
         this should only be used in scripts that only expect TestEnv args.
         """
-        # Subparsers cannot let us specify --kind=android, at best we could
-        # have --android which is lousy. Instead, use a first parser to figure
-        # out the target kind, then create a new parser for that specific kind.
-        kind_parser = argparse.ArgumentParser(
-            # Disable the automatic help to not catch e.g. ./script.py -k linux -h
-            add_help=False,
+        parser = argparse.ArgumentParser(
             formatter_class=argparse.RawDescriptionHelpFormatter,
             description=textwrap.dedent(
                 """
                 Extra arguments differ depending on the value passed to 'kind'.
                 Try e.g. "--kind android -h" to see the arguments for android targets.
                 """))
 
-        kind_parser.add_argument(
-            "--kind", "-k", choices=["android", "linux", "host"],
-            help="The kind of target to create")
 
-        # Add a self-managed help argument, see why below
-        kind_parser.add_argument("--help", "-h", action="store_true")
+        parser = argparse.ArgumentParser()
```

Comment:
So overall I'm okay with the single parser as it's much easier to maintain and develop, but I miss the contextual help that made things crystal clear.

It's not coming back, so could we augment the CLI help to give a few examples?

Reply:
sure, I'll add some